viewing_party/party.py

```diff
@@ -1,23 +1,192 @@
+    """ Return the most-watched genre in user_data.
+    
+    Keyword arguments:
+    user_data -- a dictionary with a "watched" key with a list value
+    """
+    all_genres = {}
+    most_watched_genre = None
 
+    if len(user_data["watched"]) == 0:
+        return most_watched_genre
+    else:
+        for movie in user_data["watched"]:
+            if movie["genre"] in all_genres:
+                all_genres[movie["genre"]] += 1
+            else:
+                all_genres[movie["genre"]] = 1
+        most_watched_genre = max(all_genres, key=all_genres.get)
+        return most_watched_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data: dict):
+    """Return a list of movies uniquely watched by the user.
+
+    Keyword arguments:
+    user_data -- A dictionary with "watched" and "friends" keys.      
+    """
+    friend_movies = get_friend_watched(user_data)
+
+    unique_movies = [movie for movie in user_data["watched"]
+                     if movie not in friend_movies]
+    return unique_movies
+
+def get_friends_unique_watched(user_data: dict):
+    """Return a list of movies uniquely watched by the user's friends.
+
+    Keyword arguments:
+    user_data -- A dictionary with "watched" and "friends" keys.      
+    """
+    unique_movies = []
+    friend_movies = get_friend_watched(user_data)
+    
+    # I chose this structure in cases where there were multiple if statements
+    for movie in friend_movies:
+        if movie not in unique_movies and movie not in user_data["watched"]:
+            unique_movies.append(movie)
+    return unique_movies
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data: dict):
+    """ Return a list of recommended movies based on user_data.
+    Will only return movies available on user's streaming subscriptions.
+
+    Keyword arguments:
+    user_data -- a dictionary with "subscriptions" and "friends" keys
+    """
+    friend_recommendations = get_friends_unique_watched(user_data)
+    
+    recs = [movie for movie in friend_recommendations
+            if movie["host"] in user_data["subscriptions"]]
+    return recs
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data: dict):
+    """ Return a list of recommended movies based on user_data.
+    Will select movies that match user's most frequently watched genre.
+
+    Keyword arguments:
+    user_data -- a dictionary with "friends" and "watched" keys
+    """
+    fave_genre = get_most_watched_genre(user_data)
+    friend_recs = get_friends_unique_watched(user_data)
+    
+    recs = [movie for movie in friend_recs if movie["genre"] == fave_genre]
+    return recs

```

Comment:
Great job with the comprehension again! I've noticed a few cases where you're defining a variable, then returning it immediately. You can just return the comprehension although there might be instances in which you want to assign and use it before returning it.